src/main/sensors/barometer.c

```diff
@@ -174,11 +174,18 @@ int32_t baroCalculateAltitude(void)
 
 void performBaroCalibrationCycle(void)
 {
+    static int32_t savedGroundPressure=0;
+
     baroGroundPressure -= baroGroundPressure / 8;
     baroGroundPressure += baroPressureSum / PRESSURE_SAMPLE_COUNT;
     baroGroundAltitude = (1.0f - powf((baroGroundPressure / 8) / 101325.0f, 0.190295f)) * 4433000.0f;
 
-    calibratingB--;
+    if (baroGroundPressure==savedGroundPressure)
```

Comment:
Well, maybe, but I don't think so. baroGroundPressure is a (fake) moving average with 8 samples so it will take more than two equal samples to stop the calibration, wouldn't it?